integration/spark2/src/main/scala/org/apache/spark/util/SparkTypeConverter.scala

```diff
@@ -97,16 +97,30 @@ private[spark] object SparkTypeConverter {
   def getStructChildren(table: CarbonTable, dimName: String): String = {
     table.getChildren(dimName).asScala.map(childDim => {
       childDim.getDataType.getName.toLowerCase match {
-        case "array" => s"${
+        case "array" => if (table.isTransactionalTable) {s"${
```

Comment:
Could explain in comments why this many checks are required for non transactional table? 
Better write another method rather than keeping so many if checks

Reply:
Written the comment of why transactional and no transactional behaviour will be different.